### ISSUE TITLE
CI tests are failing. (Probably due to new numpy v2.0.0)

### ISSUE THREAD

**Issue by abhisrkckl** [nanograv/PINT#1783]

**Comment by dlakaplan** [comment]
Should we limit things to <2.0 for now?  I think other libraries (like astropy) will also need to update.

**Comment by abhisrkckl** [comment]
I think that's the sensible thing to do to avoid breaking things.

**Comment by abhisrkckl** [comment]
@paulray @scottransom What do you think?

**Comment by dlakaplan** [comment]
One issue is around longdoubles.  I see behavior like:
```
In [1]: np.longdouble('41498.0')
Out[1]: np.longdouble('41498.0')
```
with numpy 2.0, while previously it returned just the numerical value.  The value it returns is useable, but it confuses the float/string conversion tasks.  It's further confused by the attempt to change fortran values to normal exponentials, since the word `longdouble` gets changed to `longeouble`.  Not sure what the fix is (the fix for the second part shouldn't be hard, but it doesn't work even without that).

**Comment by dlakaplan** [comment]
Actually, I might have fixes for most of the failures.  See #1784

**Comment by scottransom** [comment]
Yeah, I think we should wait until our upstream dependencies handle Numpy 2.0.0 before we try to do too much work (but thanks to @dlakaplan for doing some initial tests!).  So I think a <2.0 limit is fine for now, but we should work on the rest of the fixes pretty soon.  (Note that for pure python packages like ours, there probably won't be many issues.  A lot of the v2.0.0 teething problems come because the C API is changing quite a bit.  So that will likely affect (or has already affected?) ERFA, and therefore much of AstroPy).

**Comment by dlakaplan** [comment]
OK, the PR I started passes all tests (I just added a CHANGELOG so it will take time for that to show up).

I agree that other changes may be needed if astropy etc change, but this works with the latest current astropy release (6.1.1).  Most of the changes needed on our end are pretty minimal (change `repr` -> `str`, use explicit `float64` in some tests) and I don't think they should change the functionality.  So perhaps we don't need to restrict the version of numpy but if PR isn't accepted we can still do that.
